### PR TITLE
Minor documentation fixes

### PR DESCRIPTION
**Pull request by iblech**
The first commit adds a short note which I believe will reduce worries in people who work with closures very often and read the Rust book for their first time.

The second commit consists solely of tiny typo fixes. In some cases, I changed "logical" quotations like

```
She said, "I like programming".
```

to

```
She said, "I like programming."
```

because the latter seems to be the prevalent style in the book.

### CODE REVIEW

**Comment by nikomatsakis** [comment]
Thanks for the suggestion! I'm assigning to @steveklabnik -- he may want to tweak the wording; I feel like this is a part of the book that gets encountered fairly early, if I'm not mistaken. Personally I'd probably change it to something like "(Note though that types do not have to be declared in closures.)"

**Comment by iblech** [comment]
Thanks for the quick reply! Indeed, I hit that part of the book early. @steveklabnik, of course you can feel free to change the wording and/or ask me to do it (as a further commit to this pull request).

File: src/doc/trpl/documentation.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ comments":
 // the "link" crate attribute is currently required for rustdoc, but normally
 // isn't needed.
 #![crate_id = "universe"]
-#![crate_type="lib"]
+#![crate_type = "lib"]
```
Comment (steveklabnik) [inline src/doc/trpl/documentation.md]: this change is unrelated and shouldn't be here

File: src/doc/trpl/match.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ match x {
 `match` takes an expression and then branches based on its value. Each *arm* of
 the branch is of the form `val => expression`. When the value matches, that arm's
 expression will be evaluated. It's called `match` because of the term 'pattern
-matching', which `match` is an implementation of.
+matching,' which `match` is an implementation of.
```
Comment (steveklabnik) [inline src/doc/trpl/match.md]: The other quote stuff is good, but this one is wrong: when we use quotes for emphasis, we don't put the period inside.

File: src/doc/trpl/method-syntax.md [hunk 3]
```diff
@@ -61,7 +61,7 @@ multiplications later, and we have our area.
 ## Chaining method calls
 
 So, now we know how to call a method, such as `foo.bar()`. But what about our
-original example, `foo.bar().baz()`? This is called 'method chaining', and we
+original example, `foo.bar().baz()`? This is called 'method chaining,' and we
```
Comment (steveklabnik) [inline src/doc/trpl/method-syntax.md]: same

File: src/doc/trpl/more-strings.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ Rust has two main types of strings: `&str` and `String`.
 
 # &str
 
-The first kind is a `&str`. This is pronounced a 'string slice'.
+The first kind is a `&str`. This is pronounced a 'string slice.'
```
Comment (steveklabnik) [inline src/doc/trpl/more-strings.md]: same

File: src/doc/trpl/unsafe.md [hunk 5]
```diff
@@ -649,7 +649,7 @@ functionality that isn't hard-coded into the language, but is
 implemented in libraries, with a special marker to tell the compiler
 it exists. The marker is the attribute `#[lang="..."]` and there are
 various different values of `...`, i.e. various different "lang
-items".
+items."
```
Comment (steveklabnik) [inline src/doc/trpl/unsafe.md]: same, and should be single quotes actually

File: src/libcore/iter.rs [hunk 6]
```diff
@@ -2359,7 +2359,7 @@ pub struct Counter<A> {
 /// Creates a new counter with the specified start/step
 #[inline]
 #[unstable(feature = "core",
-           reason = "may be renamed or replaced by range notation adapaters")]
+           reason = "may be renamed or replaced by range notation adapters")]
```
Comment (steveklabnik) [inline src/libcore/iter.rs]: these two changes are welcome, but totally unrelated

Comment (steveklabnik) [inline src/libcore/iter.rs]: I guess it's okay to leave this here.

**Comment by steveklabnik** [comment]
Thanks so much!

Yes, we haven't even mentioned closures yet, and we do mention this behavior quite specifically, so I would prefer to leave it off. Most of the other changes are wonderful, though I do have a few points that shouldn't have been changed.

**Comment by steveklabnik** [comment]
@iblech : one week bump. Any interest in updating this PR?

File: src/doc/trpl/documentation.md (hunk 1 above)
Comment (iblech) [inline src/doc/trpl/documentation.md]: Sorry, I didn't want to create too much noise with trivial pull requests. Do you want me to open a new one for this issue?

(I don't have a clue about correct Rust coding style. But http://aturon.github.io/style/whitespace.html recommends it and I thought it would be good to have the official documentation align with it.)

File: src/doc/trpl/match.md (hunk 2 above)
Comment (iblech) [inline src/doc/trpl/match.md]: Ah, okay. I will revert this change.

File: src/doc/trpl/documentation.md (hunk 1 above)
Comment (steveklabnik) [inline src/doc/trpl/documentation.md]: it's all good! no PR is too small. That guideline is WIP, and I would say goes _against_ most style stuff, so let's just leave this for now, since it's already here.

File: src/doc/trpl/match.md (hunk 2 above)
Comment (steveklabnik) [inline src/doc/trpl/match.md]: no worries. i'm working on the comment doc style RFC, it just hasn't been merged, so you can't know. :/

**Comment by iblech** [comment]
Sorry for not returning to you earlier, I was busy with other stuffs. Also thank you for the detailed comments!

Are you sure about the removal of the parenthetical remark "(For closures, i.e. unnamed functions, types do not have to be declared.)"? It could be softed down, like "(If you already know closures, i.e. unnamed functions, then note that types do not have to be declared in those.)" This way I think it won't confuse people unfamiliar with closures, while greatly relieving folks like Haskell programmers.

**Comment by steveklabnik** [comment]
It's all good! 

Yes. We haven't covered closures _at all_ yet, and so introducing the concept without explanation is a distraction. Closures are something that's becoming more and more popular, but I've taken a lot of care to only introduce what's needed at the time that it's needed.

**Comment by iblech** [comment]
Okay, fine, I'll revert the addition of the remark then.

Could you please check my answer to your line note regarding src/doc/trpl/documentation.md? After this, I can add all the necessary commits/reverts to this pull request. :-)

**Comment by steveklabnik** [comment]
sorry, which line note?

**Comment by iblech** [comment]
Meanwhile, you already did! It's now hidden, because of the new commit.

**Comment by steveklabnik** [comment]
Ha! Yeah, so I think this is good to go. Thank you so much!

**Comment by iblech** [comment]
Great! Thanks for the assistance. :-)
